### ISSUE TITLE
[5.3.15] Null value retreive on a abstract property not mapped to a template but its getter linked to a mapped property 

### ISSUE THREAD

**Issue by sebmourier** [mikeedwards83/Glass.Mapper#371]
Hello I'll give you a code sample in order to give you the maximum information regarding my issue.

```
// Model classes

public abstract class AbstractBaseClass
{
   public abstract string MyAbstractPropertyNotMappedSitecore { get; }
}

 public class Class1MappedToASitecoreTemplate
 {
        virtual string FirstProperty { get; set; }
        virtual string SecondProperty { get; set; }
 }

// First concrete class
 public class ConcreteClass1MappedToASitecoreTemplate : AbstractBaseClass
 {
        public virtual Class1MappedToASitecoreTemplate Object1 { get; set; }

        public string MyPropertyNotMappedSitecore
        {
            get { return Object1.FirstProperty; }
        }

        public override string MyAbstractPropertyNotMappedSitecore
        {
            get { return Object1.FirstProperty; }
        }

        // Other properties mapped to a field in Sitecore
        ....
 }

// Second concrete class
 public class ConcreteClass2MappedToASitecoreTemplate : AbstractBaseClass
 {
  ....
 }


 // In the class requesting the data

 private readonly IRequestContext _requestContext;
 private readonly ISitecoreService _sitecoreService;
    ....
    IOC
    ....

 var currentItem = _requestContext.GetContextItem<Item>();
 var mappedItem = _sitecoreService.GetItem<ConcreteClass1MappedToASitecoreTemplate>(currentItem)

string valProperty = mappedItem.MyPropertyNotMappedSitecore
// valProperty = value of the FirstProperty In sitecore
string valAbstractProperty = mappedItem.MyAbstractPropertyNotMappedSitecore
// valAbstractProperty = null
```

My "true code" uses notion of inheritance and polymorphism and for the sake of simplicity here I don't really use this notions.

With Glass V4.4, valAbstractProperty  = value of the FirstProperty In sitecore
As you can see with Glass V5.3 it's no longer the case.

For the moment, my only workaround is to create models classes independant of Glass and Sitecore with nearly the sames properties and use DTO classes or something like that... wich add an another layer and plumbing code.

Am I missing something or is there a breaking changes in the V5 on this matter ?
In general, do you recommand mixing mapped properties with unmapped properties in a Glass class ?

**Comment by mikeedwards83** [comment]
Hi 

I think this may have been fixed in version 5.3.17.

See the release notes about ISSUE 370 http://www.glass.lu/Mapper/Releases

Mike

**Comment by sebmourier** [comment]
Hi,

You're correct, v5.317 has fixed my problem.
Thanks,

Sebastien
